cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -34,40 +35,28 @@ namespace detail {
     /// @brief Implementation class that holds the state associated with a
     /// single async, unary call.
     ///
-    /// Two different object participate in shared ownership of instances of
-    /// this class: the user-facing unary_call object and the
-    /// response_sent_tag helper struct.
+    /// Potentually three different objects participate in shared ownership
+    /// of instances of this class: the user-facing \ref unary_call and \ref
+    /// shared_unary_call objects, as well as this object itself.
     ///
-    /// To ensure that it stays alive while sending a response,
-    /// unary_call_impl creates a response_sent_tag with a shared_ptr to
-    /// itself that it enques in the completion queue. When that sending of
-    /// the response is done and it is dequeued from the completion queue,
-    /// response_sent_tag clears its shared_ptr to the unary_call_impl.
-    ///
-    /// The user-facing unary_call object also has a shared_ptr to an
-    /// instance of unary_call_impl to keep the state alive while the call
-    /// object is itself alive.
+    /// To ensure that this instance stays alive while sending a response,
+    /// unary_call_impl creates itself with a ref count of 1, opposed to the
+    /// usual 0 that many boost::intrusive_ptr implementations use. When the
+    /// notification that the response has been sent is dequeued from the
+    /// completion queue, %invoke() calls %Release() on itself, decrementing
+    /// the ref count, and allowing the remaining unary_call and
+    /// shared_unary_call objects, if any, to control lifetime.
     template <typename TRequest, typename TResponse>
-    struct unary_call_impl final
-        : std::enable_shared_from_this<unary_call_impl<TRequest, TResponse>>,
-          io_manager_tag
+    class unary_call_impl final : io_manager_tag
```

Comment:
Should this be derived as `public`?

Reply:
No.

Reply:
OK, then probably better to explicitly mark as `private`.